app/models/manageiq/providers/vmware/infra_manager/inventory/collector.rb

```diff
@@ -32,7 +32,7 @@ def stop(join_timeout = 2.minutes)
     return if join_timeout.nil?
 
     # The WaitOptions for WaitForUpdatesEx call sets maxWaitSeconds to 60 seconds
-    vim_thread&.join(join_timeout)
+    vim_thread&.join(join_timeout) if vim_thread&.alive?
```

Comment:
```suggestion
    vim_thread&.join(join_timeout) if running?
```

Reply:
Actually this `if running?` isn't doing anything because `thread.join` on a dead thread doesn't block anyway